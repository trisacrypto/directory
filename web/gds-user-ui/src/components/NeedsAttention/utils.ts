export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
