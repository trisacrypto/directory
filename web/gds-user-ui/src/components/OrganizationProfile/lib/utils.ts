export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
