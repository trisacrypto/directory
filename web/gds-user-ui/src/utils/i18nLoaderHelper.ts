import { i18n } from '@lingui/core';
import { en, fr, de, zh, ja, ru } from 'make-plural/plurals';
import { Locales } from 'types/type';

export const DEFAULT_LOCALE: Locales = 'en';

export async function dynamicActivate(locale: string) {
  const { messages } = await import(`../locales/${locale}/messages`);

  i18n.load(locale, messages);
  i18n.loadLocaleData({
    en: { plurals: en },
    de: { plurals: de },
    fr: { plurals: fr },
    zh: { plurals: zh },
    ja: { plurals: ja }
  });
  i18n.activate(locale);
}
