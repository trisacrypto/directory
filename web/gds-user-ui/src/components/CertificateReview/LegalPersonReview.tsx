import React, { FC, useEffect } from 'react';
import {
  Stack,
  Box,
  Text,
  Heading,
  Table,
  Tbody,
  Tr,
  Td,
  Button,
  Tag,
  Divider
} from '@chakra-ui/react';
import { colors } from 'utils/theme';
import { useDispatch, useSelector, RootStateOrAny } from 'react-redux';
import { getStepData } from 'utils/utils';
import { loadDefaultValueFromLocalStorage, TStep } from 'utils/localStorageHelper';
import useCertificateStepper from 'hooks/useCertificateStepper';
import { getNameIdentiferTypeLabel } from 'constants/name-identifiers';
import { getNationalIdentificationLabel } from 'constants/national-identification';
import { COUNTRIES } from 'constants/countries';
interface LegalReviewProps {}
// NOTE: need some clean up.
const isValidIvmsAddress = (address: any) => {
  if (address) {
    return !!(address.country && address.address_type);
  }
  return false;
};

const hasAddressLine = (address: any) => {
  if (isValidIvmsAddress(address)) {
    return Array.isArray(address.address_line) && address.address_line.length > 0;
  }
  return false;
};

const hasAddressField = (address: any) => {
  if (isValidIvmsAddress(address) && !hasAddressLine(address)) {
    return !!(address.street_name && (address.building_number || address.building_name));
  }
  return false;
};

const hasAddressFieldAndLine = (address: any) => {
  if (hasAddressField(address) && hasAddressLine(address)) {
    console.warn('[ERROR]', 'cannot render address');
    return true;
  }
  return false;
};

export const renderLines = (address: any) => (
  <address data-testid="addressLine">
    {address.address_line.map(
      (addressLine: any, index: number) => addressLine && <div key={index}>{addressLine} </div>
    )}
    <div>{address?.country}</div>
  </address>
);

export const renderField = (address: any) => (
  <address data-testid="addressField">
    {address.sub_department ? (
      <>
        {address?.sub_department} <br />
      </>
    ) : null}
    {address.department ? (
      <>
        {address?.department} <br />
      </>
    ) : null}
    {address.building_number} {address?.street_name}
    <br />
    {address.post_box ? (
      <>
        P.O. Box: {address?.post_box} <br />
      </>
    ) : null}
    {address.floor || address.room || address.building_name ? (
      <>
        {address?.floor} {address?.room} {address?.building_name} <br />
      </>
    ) : null}
    {address.district_name ? (
      <>
        {address?.district_name} <br />
      </>
    ) : null}
    {address.town_name || address.town_location_name || address.country_sub_division ? (
      <>
        {address?.town_name} {address?.town_location_name} {address?.country_sub_division}{' '}
        {address?.post_code} <br />
      </>
    ) : null}
    {address?.country}
  </address>
);

const renderAddress = (address: any) => {
  if (hasAddressFieldAndLine(address)) {
    console.warn('[ERROR]', 'invalid address with both fields and lines');
    return <div>Invalid Address</div>;
  }

  if (hasAddressLine(address)) {
    return renderLines(address);
  }

  if (hasAddressField(address)) {
    return renderField(address);
  }

  console.warn('[ERROR]', 'could not render address');
  return <div>Unparseable Address</div>;
};

const LegalPersonReview: React.FC<LegalReviewProps> = (props) => {
  const { jumpToStep } = useCertificateStepper();
  const steps: TStep[] = useSelector((state: RootStateOrAny) => state.stepper.steps);
  const [legalPerson, setLegalPerson] = React.useState<any>({});

  useEffect(() => {
    const getStepperData = loadDefaultValueFromLocalStorage();
    const stepData = {
      ...getStepperData.entity
    };
    setLegalPerson(stepData);
  }, [steps]);
  return (
    <Box
      border="1px solid #DFE0EB"
      fontFamily={'Open Sans'}
      color={'#252733'}
      bg={'white'}
      fontSize={18}
      p={5}
      px={5}>
      <Stack>
        <Box display={'flex'} justifyContent="space-between" pt={4} ml={5}>
          <Heading fontSize={20}>Section 2: Legal Person</Heading>
          <Button
            bg={colors.system.blue}
            color={'white'}
            height={'34px'}
            onClick={() => jumpToStep(2)}
            _hover={{
              bg: '#10aaed'
            }}>
            {' '}
            Edit{' '}
          </Button>
        </Box>
        <Stack fontSize={18}>
          <Table
            sx={{
              'td:nth-child(2),td:nth-child(3)': { fontWeight: 'bold', paddingLeft: 0 },
              Tr: { borderStyle: 'hidden' }
            }}>
            <Tbody
              sx={{
                '*': {
                  fontSize: '1rem'
                }
              }}>
              <Tr>
                <Td fontSize={'1rem'} fontWeight="bold" colSpan={3}>
                  <Text mb={1}>Name Identifiers</Text>
                  <Divider />
                </Td>
              </Tr>
              <Tr>
                <Td fontStyle={'italic'} pt={0}>
                  The name and type of name by which the legal person is known.
                </Td>
                <Td>
                  <Tr>
                    {legalPerson.name?.name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} border="none">
                              {nameIdentifier.legal_person_name || 'N/A'}
                            </Td>
                            <Td paddingLeft={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </Tr>
                  <>
                    {legalPerson.name?.local_name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} pt={0} border="none">
                              {nameIdentifier.legal_person_name}
                            </Td>
                            <Td paddingLeft={0} pt={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </>
                  <>
                    {legalPerson.name?.phonetic_name_identifiers?.map(
                      (nameIdentifier: any, index: number) => {
                        return (
                          <React.Fragment key={index}>
                            <Td paddingLeft={0} pt={0} border="none">
                              {nameIdentifier.legal_person_name}
                            </Td>
                            <Td paddingLeft={0} pt={0} border="none">
                              {getNameIdentiferTypeLabel(
                                nameIdentifier.legal_person_name_identifier_type
                              )}
                            </Td>
                          </React.Fragment>
                        );
                      }
                    )}
                  </>
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Addresses</Td>
                <Td>
                  <Tr>
                    <Td paddingLeft={0} pt={0}>
                      {legalPerson?.geographic_addresses?.map((address: any, index: number) => (
                        <React.Fragment key={index}>{renderAddress(address)}</React.Fragment>
                      ))}
                      {/* {legalPerson?.geographic_addresses?.[0]?.address_line.map(
                        (line: any, i: any) => {
                          return <Text key={i}>{line}</Text>;
                        }
                      )} */}
                    </Td>
                    <Td pt={0}>{legalPerson?.geographic_addresses?.[0] && 'Legal Person'}</Td>
                  </Tr>
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Country of Registration</Td>
                <Td paddingLeft={0} pt={0}>
                  {(COUNTRIES as any)[legalPerson?.country_of_registration] || 'N/A'}
                </Td>
              </Tr>
              <Tr>
                <Td fontSize={'1rem'} fontWeight="bold" colSpan={3}>
                  <Text mb={1}>National Identification</Text>
                  <Divider />
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Identification Number</Td>
                <Td paddingLeft={0}>{legalPerson?.national_identification?.national_identifier}</Td>
              </Tr>
              <Tr>
                <Td pt={0}>Identification Type</Td>
                <Td pt={0}>
                  <Tag color={'white'} bg={'blue.400'} size={'lg'}>
                    {getNationalIdentificationLabel(
                      legalPerson?.national_identification?.national_identifier_type
                    )}
                  </Tag>
                </Td>
              </Tr>
              <Tr>
                <Td>Country of Issue</Td>
                <Td>
                  {(COUNTRIES as any)[legalPerson?.national_identification?.country_of_issue] ||
                    'N/A'}
                </Td>
              </Tr>
              <Tr>
                <Td pt={0}>Reg Authority</Td>
                <Td pt={0}>
                  {legalPerson?.national_identification?.registration_authority || 'N/A'}
                </Td>
              </Tr>
            </Tbody>
          </Table>
        </Stack>
      </Stack>
    </Box>
  );
};
LegalPersonReview.defaultProps = {
  data: {}
};
export default LegalPersonReview;
