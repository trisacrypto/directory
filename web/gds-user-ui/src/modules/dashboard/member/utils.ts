import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

