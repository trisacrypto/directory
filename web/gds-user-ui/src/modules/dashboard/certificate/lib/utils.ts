import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
