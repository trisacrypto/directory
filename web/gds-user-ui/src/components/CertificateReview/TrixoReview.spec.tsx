import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { act, render } from 'utils/test-utils';
import LegalPersonReview from './LegalPersonReview';
import TrixoReview from './TrixoReview';

const defaultValues = {
  trixo: {
    primary_national_jurisdiction: 'AF',
    primary_regulator: '',
    other_jurisdictions: [],
    financial_transfers_permitted: 'no',
    has_required_regulatory_program: 'no',
    conducts_customer_kyc: false,
    kyc_threshold: 0,
    kyc_threshold_currency: 'USD',
    must_comply_travel_rule: false,
    applicable_regulations: [{ name: 'FATF Recommendation 16' }],
    compliance_threshold: 3000,
    compliance_threshold_currency: 'USD',
    must_safeguard_pii: false,
    safeguards_pii: false
  }
};

describe('<TrixoReview />', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
  });

  beforeEach(() => {
    localStorage.clear();
  });

  it('should match snapshot', () => {
    localStorage.setItem('certificateForm', JSON.stringify(defaultValues));

    const { container } = render(<TrixoReview />);

    expect(localStorage.getItem).toHaveBeenCalledWith('certificateForm');
    expect(Object.keys(localStorage.__STORE__).length).toBe(1);

    expect(container).toMatchSnapshot();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
