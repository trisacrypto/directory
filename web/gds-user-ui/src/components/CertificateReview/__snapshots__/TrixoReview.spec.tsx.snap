// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrixoReview /> should match snapshot 1`] = `
.emotion-0 {
  border: 1px solid #DFE0EB;
  font-family: Open Sans;
  color: #252733;
  background: var(--chakra-colors-white);
  font-size: 18px;
  padding: var(--chakra-space-5);
  -webkit-padding-start: var(--chakra-space-5);
  padding-inline-start: var(--chakra-space-5);
  -webkit-padding-end: var(--chakra-space-5);
  padding-inline-end: var(--chakra-space-5);
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-top: var(--chakra-space-4);
  margin-left: 0px;
}

.emotion-3 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: 20px;
  line-height: 1.33;
  margin-bottom: 2rem;
}

@media screen and (min-width: 48em) {
  .emotion-3 {
    line-height: 1.2;
  }
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: 34px;
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: blue;
  color: var(--chakra-colors-white);
}

.emotion-4:focus,
.emotion-4[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-4[disabled],
.emotion-4[aria-disabled=true],
.emotion-4[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-4:hover,
.emotion-4[data-hover] {
  background: #10aaed;
}

.emotion-4:active,
.emotion-4[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
}

.emotion-5>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-6 {
  font-variant-numeric: lining-nums tabular-nums;
  border-collapse: collapse;
  width: var(--chakra-sizes-full);
}

.emotion-6 td:nth-child(2),
.emotion-6 td:nth-child(3) {
  font-weight: var(--chakra-fontWeights-semibold);
}

.emotion-6 td:nth-child(2) {
  max-width: 75%;
}

.emotion-6 Tr {
  border-style: hidden;
}

.emotion-6 td:first-child {
  width: 50%;
  padding-left: 1rem;
}

.emotion-6 td {
  border-bottom: var(--chakra-borders-none);
  -webkit-padding-start: 0px;
  padding-inline-start: 0px;
  padding-top: var(--chakra-space-2-5);
  padding-bottom: var(--chakra-space-2-5);
}

.emotion-9 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-9[data-is-numeric=true] {
  text-align: end;
}

.emotion-20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  font-weight: var(--chakra-fontWeights-medium);
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  min-height: 1.25rem;
  min-width: 1.25rem;
  font-size: var(--chakra-fontSizes-xs);
  -webkit-padding-start: var(--chakra-space-2);
  padding-inline-start: var(--chakra-space-2);
  -webkit-padding-end: var(--chakra-space-2);
  padding-inline-end: var(--chakra-space-2);
  border-radius: var(--chakra-radii-md);
  background: var(--chakra-colors-red-100);
  color: var(--chakra-colors-red-800);
}

.emotion-20:focus,
.emotion-20[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-21 {
  line-height: 1.2;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-weight: var(--chakra-fontWeights-bold);
}

.emotion-25 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
  font-weight: var(--chakra-fontWeights-bold);
  background: #E5EDF1;
  padding-left: 1rem!important;
}

.emotion-25[data-is-numeric=true] {
  text-align: end;
}

.emotion-42 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
  padding-left: 0px;
}

.emotion-42[data-is-numeric=true] {
  text-align: end;
}

.emotion-53 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: var(--chakra-space-1);
}

.emotion-53[data-is-numeric=true] {
  text-align: end;
}

.emotion-63 {
  text-align: start;
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-4);
  padding-bottom: var(--chakra-space-4);
  line-height: var(--chakra-lineHeights-5);
  border-bottom: var(--chakra-borders-1px);
  border-color: var(--chakra-colors-gray-100);
  background: #E5EDF1;
  font-weight: var(--chakra-fontWeights-bold);
  padding-left: 1rem!important;
}

.emotion-63[data-is-numeric=true] {
  text-align: end;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="chakra-stack emotion-1"
    >
      <div
        class="emotion-2"
      >
        <h2
          class="chakra-heading emotion-3"
        >
          Section 5: TRIXO Questionnaire
        </h2>
        <button
          class="chakra-button emotion-4"
          type="button"
        >
          Edit
        </button>
      </div>
      <div
        class="chakra-stack emotion-5"
      >
        <table
          class="chakra-table emotion-6"
          role="table"
        >
          <tbody
            class="emotion-7"
          >
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Primary National Jurisdiction
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                N/A
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Name of Primary Regulator
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                N/A
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Other Jurisdictions
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                N/A
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Is your organization permitted to send and/or receive transfers of virtual assets in the jurisdictions in which it operates?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  />
                </span>
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-25"
                colspan="3"
                role="gridcell"
              >
                Customer Due Diligence (CDD) & Travel Rule Policies
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Does your organization have a programme that sets minimum Anti-Money Laundering (AML), Countering the Financing of Terrorism (CFT), Know your Counterparty/Customer Due Diligence (KYC/CDD) and Sanctions standards per the requirements of the jurisdiction(s) regulatory regimes where it is licensed/approved/registered?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  >
                    N/A
                  </span>
                </span>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Does your organization conduct Know your KYC/CDD before permitting its customers to send/receive virtual asset transfers?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  >
                    No
                  </span>
                </span>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                At what threshold and currency does your organization conduct KYC checks?
              </td>
              <td
                class="emotion-42"
                role="gridcell"
              >
                <p
                  class="chakra-text emotion-7"
                >
                  $0.00
                   
                </p>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Is your organization required to comply with the application of the Travel Rule standards in the jurisdiction(s) where it is licensed/approved/registered?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  >
                    NO
                  </span>
                </span>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Applicable Regulations
              </td>
              <td
                class="emotion-53"
                role="gridcell"
              />
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                What is the minimum threshold for Travel Rule compliance?
              </td>
              <td
                class="emotion-42"
                role="gridcell"
              >
                <p
                  class="chakra-text emotion-7"
                >
                  $0.00
                   
                  USD
                </p>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-63"
                colspan="3"
                role="gridcell"
              >
                Data Protection Policies
              </td>
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Is your organization required by law to safeguard Personally Identifiable Information (PII)?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  >
                    NO
                  </span>
                </span>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
            <tr
              class="emotion-7"
              role="row"
            >
              <td
                class="emotion-9"
                role="gridcell"
              >
                Does your organization secure and protect Personally Identifiable Information (PII), including Personally Identifiable Information (PII) received from other VASPs under the Travel Rule?
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              >
                <span
                  class="emotion-20"
                >
                  <span
                    class="emotion-21"
                  >
                    NO
                  </span>
                </span>
              </td>
              <td
                class="emotion-9"
                role="gridcell"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
